### PR TITLE
Make read entry request recyclable

### PR DESCRIPTION
**Pull request by codelipenghui**
Descriptions of the changes in this PR:

### Motivation

(Explain: why you're making that change, what is the problem you're trying to solve)

### Changes

(Describe: what changes you have made)

Master Issue: #<master-issue-number>

> ---
> In order to uphold a high standard for quality for code contributions, Apache BookKeeper runs various precommit
> checks for pull requests. A pull request can only be merged when it passes precommit checks.
>
> ---
> Be sure to do all of the following to help us incorporate your contribution
> quickly and easily:
>
> If this PR is a BookKeeper Proposal (BP):
>
> - [ ] Make sure the PR title is formatted like:
>     `<BP-#>: Description of bookkeeper proposal`
>     `e.g. BP-1: 64 bits ledger is support`
> - [ ] Attach the master issue link in the description of this PR.
> - [ ] Attach the google doc link if the BP is written in Google Doc.
>
> Otherwise:
> 
> - [ ] Make sure the PR title is formatted like:
>     `<Issue #>: Description of pull request`
>     `e.g. Issue 123: Description ...`
> - [ ] Make sure tests pass via `mvn clean apache-rat:check install spotbugs:check`.
> - [ ] Replace `<Issue #>` in the title with the actual Issue number.
> 
> ---

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java [hunk 1]
```diff
@@ -1170,6 +1170,10 @@ private void writeAndFlush(final Channel channel,
             });
 
             channel.writeAndFlush(request, promise);
+            if (request instanceof BookieProtocol.Request) {
```
Comment (hangc0276) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java:1173]: `request instanceof BookieProtocol.ReadRequest`?

Comment (hangc0276) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java:1173]: We need to add the release logic at the line 1154

Comment (hangc0276) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java:1173]: Do we need to move the release logic in the finally?

Comment (codelipenghui) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java:1169]: Yes move to finally is better

**Review by hangc0276 — APPROVED** [review]

**Comment by codelipenghui** [comment]
close the PR since it has been merged to the apache/bookkeeper

**Pull request closed without merging**